### DISCUSSION TITLE
icon color on input type date, time + dark mode

### DISCUSSION THREAD

**Discussion by pythonormen** [zauberzeug/nicegui#1037] [Q&A]
### Question

Hi,
When using Quasar input type date and time, then the icon is dark even though dark mode is enabled. It's the same on their site but maybe there is a way to change / manage the icon color?

Thank you

```py
from nicegui import ui

with ui.row():
    ui.input(label='Date').props('type=date').classes("w-28")
    ui.input(label='Time').props('type=time').classes("w-20")

dark = ui.dark_mode()
ui.label('Switch mode:')
with ui.row():
    ui.button('Dark', on_click=dark.enable)
    ui.button('Light', on_click=dark.disable)

ui.run(host="127.0.0.1")
```

<img width="245" alt="image" src="https://github.com/zauberzeug/nicegui/assets/125542430/73209683-4910-468d-a5fc-e022d9c71c02">

<img width="260" alt="image" src="https://github.com/zauberzeug/nicegui/assets/125542430/2a48db71-977e-4fb0-bb18-4889eec6fe2d">

**Comment by falkoschindler** [comment]
Interesting bug, @pythonormen! Since it happens on the official Quasar website as well, we should check if there's a ticket on their side and maybe file a new issue.

Meanwhile, there's probably a way to workaround the problem, maybe with `ui.add_head_html` and some custom CSS. But I'm struggling to find the icons in the DOM.

**Comment by pythonormen** [comment]
Thanks, just if you had something on top of your mind. I'll adjust to some other Qinput field and look into the Quasar forum.

While looking at this I noticed that Safari sets some sort of default date / time while it's yyyy-mm-dd on Chrome and Firefox. So might as well need to change any way :) 

<img width="279" alt="image" src="https://github.com/zauberzeug/nicegui/assets/125542430/f60c4e13-68d6-40eb-9479-8afaf54be348">

**Comment by pythonormen** [comment]
Basically using the ones from the documentation now but any idea why it's not possible to clear the values if its entered manually (only works to clear if selected with the popup)?

```py
from nicegui import ui

async def clear_inputs():
    input_date.value = ''
    input_time.value = ''

with ui.row():
    with ui.input('Date').props('mask="####-##-##"').classes("w-28") as input_date:
        with input_date.add_slot('append'):
            ui.icon('edit_calendar', size='xs').on('click', lambda: menu_date.open())
        with ui.menu() as menu_date:
            ui.date().props('today-btn').bind_value(input_date)

    with ui.input('Time').props('mask="##:##"').classes("w-20") as input_time:
        with input_time.add_slot('append'):
            ui.icon('schedule', size='xs').on('click', lambda: menu_time.open())
        with ui.menu() as menu_time:
            ui.time().props('format24h').bind_value(input_time)

ui.button(on_click=clear_inputs).props('icon=clear')

ui.run(host="127.0.0.1")
```

**Reply by falkoschindler** [reply]
Oh wow, this looks like a nasty bug! I opened a new ticket #1071 to track this issue.
